models/intermediate/field_history/int_jira__issue_field_history.sql

```diff
@@ -12,17 +12,42 @@ fields as (
 
 ), 
 
+{% if var('jira_using_teams', True) %}
+teams as ( 
+
+    select * 
+    from {{ ref('stg_jira__team') }} 
+),
+{% endif %}
+
 joined as (
   
   select
-    field_history.*,
-    lower(fields.field_name) as field_name
+    field_history.field_id,
+    field_history.issue_id,
+    field_history.updated_at,
+    {% if var('jira_using_teams', True) %}
+    -- if the field is 'team', we want to replace the value with the team name
+    -- otherwise, just use the value as is
+    case when lower(fields.field_name) = 'team' then teams.team_name
+         else field_history.field_value end as field_value,
+    {% else %}
```

Comment:
The logic of replacing team field values (the hexadecimal ids you were seeing) is actually being produced in the `jira__daily_issue_field_history` [here](https://github.com/fivetran/dbt_jira/pull/149/files#diff-5a7efb95d836e89cb024348365dcd910e4d29865b3a99afc23b9a13351a4aa5bR154-R158) and `jira__timestamp_issue_field_history` [here](https://github.com/fivetran/dbt_jira/pull/149/files#diff-0000cae7ec3ed0c3541b4923d901a785a9e27d23071dd9611ef85968d053e91fR91-R93), as it joins to `teams` and gets the data as we're looking. 

I was able to run this model with some small tweaks to the above models and get team models. If you see the same, I think you're safe to revert the changes you made in this model.